cmd/client/lib/cli.go

```diff
@@ -0,0 +1,144 @@
+package lib
+
+import (
+	"context"
+	"encoding/hex"
+	nhttp "net/http"
+	"os"
+	"path"
+	"strconv"
+
+	"github.com/drand/drand/chain"
+	"github.com/drand/drand/client"
+	"github.com/drand/drand/client/grpc"
+	"github.com/drand/drand/client/http"
+	"github.com/drand/drand/log"
+	"github.com/drand/drand/lp2p"
+	gclient "github.com/drand/drand/lp2p/client"
+
+	"github.com/google/uuid"
+	bds "github.com/ipfs/go-ds-badger2"
+	pubsub "github.com/libp2p/go-libp2p-pubsub"
+	ma "github.com/multiformats/go-multiaddr"
+	"github.com/urfave/cli/v2"
+)
+
+var (
+	// ClientFlags is a list of common flags for client creation
+	ClientFlags = []cli.Flag{
+		&cli.StringSliceFlag{
+			Name:  "url",
+			Usage: "root URL(s) for fetching randomness",
+		},
+		&cli.StringFlag{
+			Name:    "grpc-connect",
+			Usage:   "host:port to dial a gRPC randomness provider",
+			Aliases: []string{"connect"},
+		},
+		&cli.StringFlag{
+			Name:  "cert",
+			Usage: "Path to a file containing gRPC transport credentials of peer",
+		},
+		&cli.StringFlag{
+			Name:  "hash",
+			Usage: "The hash (in hex) for the chain to follow",
+		},
+		&cli.BoolFlag{
+			Name:  "insecure",
+			Usage: "Allow autodetection of the chain information",
+		},
+		&cli.StringSliceFlag{
+			Name:  "relay",
+			Usage: "relay peer multiaddr(s) to connect with",
+		},
+		&cli.IntFlag{
+			Name:  "port",
+			Usage: "Local port for client to bind to, when connecting to relays",
+		},
+	}
+)
+
+// Create builds a client, and can be invoked from a cli action supplied
+// with ClientFlags
+func Create(c *cli.Context, withInstrumentation bool, opts ...client.Option) (client.Client, error) {
+	if c.IsSet("grpc-connect") {
+		return grpc.New(c.String("grpc-connect"), c.String("cert"), c.IsSet("insecure"))
+	}
+	var hash []byte
+	var err error
+	if c.IsSet("hash") {
+		hash, err = hex.DecodeString(c.String("hash"))
+		if err != nil {
+			return nil, err
+		}
+		opts = append(opts, client.WithChainHash(hash))
+	}
+	if c.IsSet("insecure") {
```

Comment:
We should use `c.Bool("insecure")` to get the boolean value - I had a problem with Travis when he set the flags as `--insecure=false` for example and I was only checking if that was set or not as here.